integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/CreatePreAggregateTableCommand.scala

```diff
@@ -94,10 +95,22 @@ case class CreatePreAggregateTableCommand(
       dmproperties.foreach(f => childSchema.getProperties.put(f._1, f._2))
       // updating the parent table about child table
       PreAggregateUtil.updateMainTable(parentDbName, parentTableName, childSchema, sparkSession)
-      val loadAvailable = PreAggregateUtil.checkMainTableLoad(parentTable)
-      if (loadAvailable) {
-        sparkSession.sql(
-          s"insert into ${ tableModel.databaseName }.${ tableModel.tableName } $queryString")
+      val availableLoads = PreAggregateUtil.checkMainTableLoad(parentTable)
+      if (availableLoads) {
+        val headers = childSchema.getChildSchema.getListOfColumns.asScala.map(_.getColumnName)
+          .mkString(",")
+        val childDataFrame = Dataset.ofRows(sparkSession, new CarbonSpark2SqlParser()
+          .parse(s"insert into ${ tableModel.databaseName }.${
+            tableModel.tableName} $queryString"))
```

Comment:
why it is not just `queryString`? why insertinto required here as you are already using load command